stlab/concurrency/utility.hpp

```diff
@@ -86,6 +86,40 @@ T blocking_get(future<T> x) {
     return result;
 }
 
+template <typename T>
+boost::optional<T> blocking_get(future<T> x,const std::chrono::nanoseconds& timeout) {
+    boost::optional<T> result;
+    std::exception_ptr error;
+
+    bool set{false};
+    std::condition_variable condition;
+    std::mutex m;
+    auto hold = std::move(x).recover([&](auto&& r) {
+        {
+            std::unique_lock<std::mutex> lock(m);
+            if (r.error())
+                error = std::forward<decltype(r)>(r).error().value();
+            else
+                result = std::forward<decltype(r)>(r).get_try().value();
+            set = true;
+        }
+        condition.notify_one();
+    });
+    
+    std::unique_lock<std::mutex> lock{m};
+    while (!set) {
+        if(condition.wait_for(lock, timeout) == std::cv_status::timeout) {
+            hold.reset();
+            return {};
```

Comment:
I would not return {} here, but result, because of NRVO.

Reply:
Ah good point!

Reply:
Done.